nabla-indexer/test/indexer_tests.ts

```diff
@@ -0,0 +1,1067 @@
+
+      assertTrue(indexerRouters !== undefined);
+
+      router = indexerRouters.find((router) => router.id === address(instance.router))!;
+      assertTrue(router !== undefined);
+
+      backstopPool = router.backstopPool.find((backstopPool) => backstopPool.id === address(instance.backstop))!;
+      assertTrue(backstopPool !== undefined);
+      assertEq(backstopPool.reserves, "12000000000000");
+      assertEq(backstopPool.totalSupply, "12000000000000");
+    },
+
+    async testSwaps() {
+      const instance = await createNablaInstance(unit(10), [unit(11), unit(12)]);
+      let indexerRouters = (await readIndexerUntil(
+        (routers) => routers.find((router) => router.id === address(instance.router)) !== undefined,
+        3000
+      ))!;
+      assertTrue(indexerRouters !== undefined);
+
+      await instance.router.swapExactTokensForTokens(
+        unit(2),
+        0,
+        [address(instance.assets[1]), address(instance.assets[2])],
+        tester,
+        MAX_UINT256
+      );
+
+      indexerRouters = (await readIndexerUntil(
+        (routers) =>
+          routers
+            .find((router) => router.id === address(instance.router))
+            ?.swapPools.find((pool) => pool.id === address(instance.swapPools[0]))?.reserve === "12997193504712",
+        3000
+      ))!;
+      assertTrue(indexerRouters !== undefined);
+
+      verifyIndexer(indexerRouters, instance, [
+        {
+          reserve: "12997193504712",
+          reserveWithSlippage: "13000000000000",
+          totalLiabilities: "11000000000000",
+          totalSupply: "11000000000000",
+          tokenName: "Token 1",
+        },
+        {
+          reserve: "10668537663526",
+          reserveWithSlippage: "10670015664770",
+          totalLiabilities: "12000000000000",
+          totalSupply: "12000000000000",
+          tokenName: "Token 2",
+        },
+      ]);
+
+      await instance.router.swapExactTokensForTokens(
+        unit(3),
+        0,
+        [address(instance.assets[2]), address(instance.assets[1])],
+        tester,
+        MAX_UINT256
+      );
+
+      indexerRouters = (await readIndexerUntil(
+        (routers) =>
+          routers
+            .find((router) => router.id === address(instance.router))
+            ?.swapPools.find((pool) => pool.id === address(instance.swapPools[0]))?.reserve === "8497760346518",
+        3000
+      ))!;
+      assertTrue(indexerRouters !== undefined);
+
+      verifyIndexer(indexerRouters, instance, [
+        {
+          reserve: "8497760346518",
+          reserveWithSlippage: "8504206392314",
+          totalLiabilities: "11000000000000",
+          totalSupply: "11000000000000",
+          tokenName: "Token 1",
+        },
+        {
+          reserve: "13668159768989",
+          reserveWithSlippage: "13670015664770",
+          totalLiabilities: "12000000000000",
+          totalSupply: "12000000000000",
+          tokenName: "Token 2",
+        },
+      ]);
+    },
+
```

Comment:
Should we check to see if the balances were modified to reflect the swap? Or is it too much coupling on the same test.

Reply:
What balances do you refer to? The pool token balance(s) and/or the LP token balance(s)? As these are not reported by the indexer and, e.g., the Nabla UI reads them directly from the smart contract, I don't actually consider this a necessity of this test suite. Is that what you are referring to?

Reply:
Is it fine if we leave that out?